test/stage2/compare_output.zig

```diff
@@ -539,4 +544,35 @@ pub fn addCases(ctx: *TestContext) !void {
             "",
         );
     }
+
+    {
+        var case = ctx.exe("wasm returns", wasi);
+
+        case.addCompareOutput(
+            \\export fn _start() u32 {
+            \\    return 42;
+            \\}
+        ,
+            "42\n",
+        );
+
+        case.addCompareOutput(
+            \\export fn _start() i64 {
+            \\    return 42;
+            \\}
+        ,
+            "42\n",
+        );
+
+        case.addCompareOutput(
+            \\export fn _start() f32 {
+            \\    return 42.0;
+            \\}
+        ,
+            // This is what you get when you take the bits of the IEE-754
+            // representation of 42.0 and reinterpret them as an unsigned
+            // integer. Guess that's a bug in wasmtime.
```

Comment:
Not sure if it's a bug: at least in wasi the main/start function always returns int